python/README.md

```diff
@@ -4,11 +4,122 @@ This library provides a Pythonic API wrapper for the reference Arrow C++
 implementation, along with tools for interoperability with pandas, NumPy, and
 other traditional Python scientific computing packages.
 
-#### Development details
+### Development details
 
 This project is layered in two pieces:
 
 * pyarrow, a C++ library for easier interoperability between Arrow C++, NumPy,
   and pandas
 * Cython extensions and pure Python code under arrow/ which expose Arrow C++
-  and pyarrow to pure Python users
\ No newline at end of file
+  and pyarrow to pure Python users
+
+#### PyArrow Installation
+These are instructions on how to install PyArrow from scratch on Linux (assuming arrow is not yet installed)
+
+1. **g++ and gcc**
+
+  Make sure the latest versions of g++ and gcc are installed.
+  ```bash
+  sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+  sudo apt-get update
+  sudo apt-get install gcc-4.9 g++-4.9
+  ```
+
+2. **cmake**
+  ```bash
+  sudo add-apt-repository ppa:george-edison55/cmake-3.x
+  sudo apt-get update
+  sudo apt-get install cmake
+  ```
+
+3. **boost**
+  ```bash
+  sudo apt-get install build-essential g++ python-dev autotools-dev libicu-dev build-essential libbz2-dev
+  wget -O boost_1_60_0.tar.gz http://sourceforge.net/projects/boost/files/boost/1.60.0/boost_1_60_0.tar.gz/download
+  tar xzvf boost_1_60_0.tar.gz
+  cd boost_1_60_0
+  ./bootstrap.sh --prefix=/usr/local
+  sudo ./b2 install
+  ```
+
+3. **miniconda (optional)**
+
+  Installing mininconda is optional but is recommended to easily install parquet-cpp and other dependencies (see below). Skip this if you prefer installing parquet-cpp from source.
+  ```bash
+  # Assumes that you are in the root of the arrow project.
+  export HOME=$(pwd)
+  # Change these if you would like to mininconda to reside in a different location.
+  export MINICONDA=$HOME/miniconda
+
+  wget -O miniconda.sh https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh
+
+  bash miniconda.sh -b -p $MINICONDA
+  export PATH="$MINICONDA/bin:$PATH"
+
+  conda update -y -q conda
+  conda config --set show_channel_urls True
+  conda config --add channels https://repo.continuum.io/pkgs/free
+  conda config --add channels conda-forge
+  conda config --add channels apache
+  conda info -a
+
+  conda install --yes conda-build jinja2 anaconda-client
+  conda install -y nomkl
+  ```
```

Comment:
Since these instructions only currently relate to the active development environment for pyarrow, I do not think they should be included. It is sufficient to state that

`conda install parquet-cpp -c apache/channel/dev`

will work for the moment (though this will eventually move to conda forge.